### PR TITLE
Em/periodic

### PR DESCRIPTION
**Pull request by medinaeder**
New pull request

### CODE REVIEW

File: examples/levelset/levelset_periodic.py
Comment (florianwechsung) [inline examples/levelset/levelset_periodic.py]: Is this a question that should be answered before merging this?

**Comment by florianwechsung** [comment]
Looks good to me, but the linter is still unhappy. Please run `make lint` to check that you follow all the python style guides

**Comment by medinaeder** [comment]
I have never "linted" before. Do I need to make changes manually? I feel like i have seen some auto linting commits before in other repos.

**Comment by florianwechsung** [comment]
There are plugins for various editors that make these changes automatically (e.g. Black or AutoPEP for vim), but usually people just make them by hand. After a while you get used to the requirements and just write pep conforming code to begin with

File: examples/levelset/levelset_periodic.py
Comment (medinaeder) [reply examples/levelset/levelset_periodic.py]: @APaganini Is this addressed by the new ElastictyInnerProduct? Or marking the plex?

Comment (APaganini) [reply examples/levelset/levelset_periodic.py]: This comment has more to do with PETSc and Firedrake then Fireshape. I'll remove it now.

**Review by APaganini — APPROVED** [review]

**Comment by florianwechsung** [comment]
Looks good. Just something to think about: instead of defining a `PeriodicControlSpace`, could the standard control space just check whether the mesh is periodic and then do the appropriate thing automatically?

**Comment by medinaeder** [comment]
Ah yeah! That is a good idea. For example, it can check if the coordinates elements of the mesh are "DG". I can make that change.

File: fireshape/innerproduct.py
Comment (APaganini) [inline fireshape/innerproduct.py]: isn't it better to use `Q.is_DG` ?

File: fireshape/control.py
Comment (APaganini) [inline fireshape/control.py:165]: if we add` else: self.V_c = self.V_r`, we can spare a few lines in `get_zero_vec` and `get_space_for_inner`

I'll let @florianwechsung  decide

**Review by APaganini — CHANGES_REQUESTED** [review]
resolve two small comments

File: fireshape/innerproduct.py
Comment (medinaeder) [reply fireshape/innerproduct.py]: That was my original thought as well. However it broke a bunch of tests because other ControlSpaces don't have that option. It didn't make sense say to extend it to the other control spaces.

Comment (APaganini) [reply fireshape/innerproduct.py]: solved

**Comment by APaganini** [comment]
Just need a final decision from @florianwechsung on my question above. Then ready to merge.

File: fireshape/control.py
Comment (florianwechsung) [reply fireshape/control.py:165]: I think its fine as is, no strong opinion here.

**Review by florianwechsung — APPROVED** [review]

**Review by APaganini — APPROVED** [review]

**Pull request merged**
